docs/content/sips/008-logging.md

```diff
@@ -0,0 +1,104 @@
+title = "SIP 008 - Runtime Logging"
+template = "main"
+date = "2022-12-09T11:36:36Z"
+---
+
+Summary: Unified runtime logging
+
+Owner: mail@etehtsea.me
+
+Created: December 09, 2022
+
+## Background
+
+## Current experience
+
+Currently, by default spin prints into `stdout` only the basic info:
+
+```
+$ spin up
+
+Serving http://127.0.0.1:3000
+Available Routes:
+  hello: http://127.0.0.1:3000/hello
+    A simple component that returns hello.
+```
+
+Application log is implicitly written into `~/.spin/<spin_app>/logs/<component_id>_std{err|out}.log`.
+The log for the spin itself has log level `ERROR` and outputs into `stderr`.
+
+To enable a full logging experience, a developer needs to provide `--follow-all` to duplicate app log into `stderr` and also `RUST_LOG=spin=<log_level>` to see the spin log in the `stderr`.
+
+
+## Proposal
+
+- Set default log level to `INFO`.
+- Set default log destination to `stdout/stderr`.
+- Provide a way for spin apps for logging (besides using `println!`/`eprintln!`).
+- Log `WARN`and `ERROR` into `stderr` and `INFO`, `DEBUG`, `TRACE` into `stdout`.
+- Log level `OFF` disables logging completely.
+
+### Goal
+
+To provide the new baseline that gives foundation for further iterative development of new features and enhancements based on the feedback.
+
+### Configuration
+
+Command-line arguments:
+
+```
+--log-level <LOG_LEVEL>
+    Describes the level of verbosity
+
+    [default: INFO]
+    [possible values: ERROR, WARN, INFO, DEBUG, TRACE, OFF]
+
+--log-timezone <LOG_TIMEZONE>
+    [possible values: local, utc]
+```
+
+`spin.toml`:
+```toml
+[variables]
+log = {
+  level: "INFO",
+  timezone: "local"
+}
+```
+
+### Format
+
+Conforms to the default format of the `tracing_subscriber`: `<rfc_3339_datetime>  <log-level> <crate::module>: <text>`.
+
+### Examples
+<TBD>
+
+
+## Future design considerations
+
+### Log format configuration
+
+Through command-line:
+
+```
+--log-format <LOG_FORMAT>
+```
+
+`spin.toml`:
```

Comment:
This is another one of those things where it's probably not an app definition concern - it might change from environment to environment depending on who or what is consuming the logs.  I agree it's a discussion for the future though, and thanks for flagging it up!